src/main/java/com/google/sps/python/heatMap.py

```diff
@@ -0,0 +1,181 @@
+def load_population_map(county_to_coordinates):
+    population_map = {}
+    line_count = 0
+    f = open(_POPULATION, "r")
+    for line in f:
+        if line_count <= 1 or (line_count in _IGNORE_LOCATIONS): # Skip invalid lines
+            line_count += 1
+            continue
+        else: 
+            row = line.split(',')
+            update_population_map(population_map, county_to_coordinates, row, line_count)
+            line_count += 1
+    return population_map
+
+
+# Checks whether the current location has a non-zero population and 
+# corresponding coordinates before adding this data into the map.
+def update_population_map(population_map, county_to_coordinates, row, line_count):
+    # Ignore locations where population = 0
+    if (row[1] == "Statewide Unallocated") or (int(row[-1]) == 0):
+        return
+    
+    county = ''.join(row[1].split()[:-1])  # Removes 'county' and 'borough' from county names
+    key = county + ", " + row[2]
+    # If we know its coordinates and haven't seen it before
+    if (key in county_to_coordinates) and (key not in population_map):
+        location_object = {
+            "location": county_to_coordinates[key],  # County coordinates
+            "weight": int(row[-1]) / 100  # County population / 100
+        }
+        population_map[key] = location_object
+    else:
+        _IGNORE_LOCATIONS.append(line_count)
+    return
+
+
+# This function accepts a file that lists the total cases of a county on each consecutive date
+# and returns a numpy array where each entry represents the weight of a given U.S.
+# county where weight = sum(newCasePerDay * index) so as to weight more recent cases heavier than
+# older ones.
+def load_weighted_cases(population_map):
+    case_matrix = []  # Each row is a county, each column is the total for given day.
+    line_count = 0
+    f = open(_CASES, "r")
+    for line in f:
+        if line_count != 0:  # skip first line
+            row = line.split(',')
+            county = ''.join(row[1].split()[:-1])  # Remove 'county' and 'borough' from county name
+            key = county + ", " + row[2]
+            
+            # If this is a new key that we should not ignore, append the available data to our matrix.
+            if key in population_map and (line_count not in _IGNORE_LOCATIONS):
+                case_matrix.append([int(num) for num in row[3:]])  # List of daily total for this county
+        line_count += 1
+    new_cases_matrix = np.diff(np.array(case_matrix))  #  New cases per day per county
+
+    # Multiplies each new case total by its index in its subarray to weight more recent cases heavier
+    weighted_cases = np.array([np.arange(len(sublist)).dot(sublist) for sublist in new_cases_matrix])  # len = num_counties
+    return weighted_cases
+
+
+coord_weight_map = load_covid_data()
+dump_file_path = 'heatWeights.txt'
+with open(dump_file_path, 'w') as outfile:
+    json.dump(coord_weight_map, outfile)
```

Comment:
wrap this the following way:
```
def main():
     # Code

if __name__ == '__main__':
    main()
```

Reasons:
1. If someone imports this file for any reason, the script will run without this. The above guard will make it so that it only runs if it's being run as an executable (like from the command line).
2. Stylistic, it clearly separates the main functionality from all the other helper importable functions.